src/s2e/Plugins/OSMonitors/OSMonitor.h

```diff
@@ -78,6 +78,12 @@ class OSMonitor : public Plugin {
 
     void dumpUserspaceMemory(S2EExecutionState *state, std::ostream &ss);
 
+    // Indicates whether the monitor is loaded and that calling its APIs is allowed.
+    // This is useful for monitors that rely on a guest kernel driver to provide
+    // information to plugins. Plugins should wait that the monitor is initialized
+    // before calling any APIs. This call can be used together with the onMonitorLoad signal.
+    virtual bool initialized() const = 0;
```

Comment:
Is there any reason why this is a virtual method? It seems that all of the `OSMonitors` (Raw, Linux and Windows) all have a `bool m_initialized` member and all of the `initialized()` methods seem to be the same (i.e. they just return `m_initialized`).

Reply:
good point